src/common/AutoComplete.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useRef, useState } from "react";
+import "./AutoComplete.scss";
+import { GoalItem } from "@src/models/GoalItem";
+
+interface AutocompleteComponentProps {
+  data: GoalItem[];
+  onSuggestionClick: (suggestion: GoalItem) => void;
+  onInputChange: (value: string) => void;
+  inputvalue: string;
+  placeholder: string;
+}
+
+const AutocompleteComponent: React.FC<AutocompleteComponentProps> = ({
+  data,
+  onSuggestionClick,
+  onInputChange,
+  inputvalue,
+  placeholder,
+}) => {
+  const [autocompleteValue, setAutocompleteValue] = useState("");
+  const [isSuggestionVisible, setIsSuggestionVisible] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const spanRef = useRef<HTMLSpanElement>(null);
+  const suggestionRef = useRef<HTMLSpanElement>(null);
+
+  // Adjust input width based on the span width
+  useEffect(() => {
```

Comment:
Use a single useEffect hook to adjust the input width based on whether suggestions are visible.  You can use a ternary operator within this useEffect to handle both cases.